### DISCUSSION TITLE
Dark Mode PR Code Changes unreadable

### DISCUSSION THREAD

**Discussion by dansiegel** [community/community#4552] [Mobile]
I love dark mode, however the existing color scheme particularly when reviewing a PR it is next to impossible to tell which lines are new, which are existing and even the red for removed lines is hard to see. This needs better contrast from the gray background to be able to more easily review the code changes.

**Comment by mxie** [comment]
Thanks for the feedback, @dansiegel! Would you mind providing a few screenshots of what you see so that we can better understand the contrast issue?

**Reply by fvdm** [reply]
The contrast is indeed too low. Especially when the screen is dimmed when you are in a dark environment. This is not limited to PR diffs.

![image](https://user-images.githubusercontent.com/93784/126456992-5fba3448-e92a-47d1-9da3-2001d6cc2699.jpeg)

![image](https://user-images.githubusercontent.com/93784/126457237-b0ac60e2-a425-4476-8928-9e7217bdf11f.jpeg)

![image](https://user-images.githubusercontent.com/93784/126457255-8ba58266-bd64-4cfc-8bf9-3453f467fb06.jpeg)

**Reply by brianlovin** [reply]
Thanks for the screenshots. We increase contrast in the left gutter to help with this problem, but it's hard to go much brighter on the background colors because they will quickly conflict with the syntax highlighting of the code itself. In that second screenshot, it could quickly become hard to read that green text on a brighter green background.

In any case, this is still something that we could polish up and try and find the right balance. In the short term, hopefully the gutter is a useful enough place to scan to make these things easier to read 😊

**Reply by fvdm** [reply]
Just updated the app. The new coloring of the line numbers helps. But there are also new problems.

- Blue markdown headers don't go well with colors.
- Same goes for light gray in comments.
- Or the red JSDoc keywords on a red/green background.
- The white text color is doing great though! 👍

![image](https://user-images.githubusercontent.com/93784/126980196-eb02245d-dfad-4293-ae07-adfb566001ee.png)

**Reply by brianlovin** [reply]
Yep, these aren't great – will work with the team on getting better diff highlighting contact here, thanks @fvdm for the screenshots.

**Reply by z0u** [reply]
Request: add a vivid, solid left border for changed lines.
![Dark mode diff in the GH app with solid green and red borders on the left edge of changed lines](https://github.com/user-attachments/assets/208dc39f-a3af-4aa8-a9ec-bbec31532b7a)

**Comment by fvdm** [comment]
Thanks @brianlovin. Would it be possible to make the background darker in the diffs? In the last screenshot the green is slightly better visible near the black background compared to the dark gray above it.

**Reply by fvdm** [reply]
For comparison, here is how darkmode diffs look in WorkingCopy app. The lines are easy to read.

![image](https://user-images.githubusercontent.com/93784/126565192-3872ff9e-a911-47bb-ac02-b383581f6280.jpeg)

**Comment by meduzen** [comment]
Cale here from the same feedback, using the Android app.

I had to switch to light mode because I couldn’t exactly see the diff for this one:

![Screenshot_20260120_035723_com.github.android.jpg](https://github.com/user-attachments/assets/b2a999ff-6b74-4ea1-9ed9-e96e493711eb)

Then thanks to this thread I had the idea ti show line numbers, making it better:

![Screenshot_20260120_035750_com.github.android.jpg](https://github.com/user-attachments/assets/95824a36-c0a2-49de-a692-391fc1becf8a)

Thou it’s okay for Markdown files, I don’t use line numbers otherwise because it takes space and lower the readability of the code (shorter lenght of line).

It’s okay-ish for line number, but still a small friction.

**Reply by donghuangya** [reply]
https://private-user-images.githubusercontent.com/9340937/537815630-95824a36-c0a2-49de-a692-391fc1becf8a.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjkzOTEyOTgsIm5iZiI6MTc2OTM5MDk5OCwicGF0aCI6Ii85MzQwOTM3LzUzNzgxNTYzMC05NTgyNGEzNi1jMGEyLTQ5ZGUtYTY5Mi0zOTFmYzFiZWNmOGEuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDEyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAxMjZUMDEyOTU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGEwN2YzNTUyMWIwNzAwMGYzZjkxZTJlZmIxNzBkYzAzN2YzZjZlMTAzMWQ3NGQ0YmQyYWY0MzU2ZGVlNmE0ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.f6NBT23C63PxzqhXy9Mgs6Z-n8BCO2QQ0Elk1vI0ofg

**Comment by matthew111870-hub** [comment]
That’s a valid concern, stronger contrast would really improve readability during PR reviews, especially for spotting diffs quickly, much like how clarity is essential when evaluating data trends in[ portland online ads and search growth](https://shorturl.at/MvhYH).

**Comment by kayson827363636636363-hub** [comment]
Totally agree, dark mode should make reviews easier, not harder, and better contrast would really help distinguish changes at a glance; clear visual cues matter just as much in code reviews as they do for[ uk pet travel and shipping experts](https://url-shortener.me/97SN) managing detailed documentation.
